### PR TITLE
Not set empty CUDA_VISIBLE_DEVICES

### PR DESCRIPTION
**Pull request by mtitov**

### CODE REVIEW

**Review by andre-merzky — APPROVED** [review]
LGTM, thanks!

File: tests/test_scheduler/test_unit/test_base.py [hunk 1]
```diff
@@ -108,8 +108,12 @@ def test_handle_cuda(self,mocked_init):
                     component._handle_cuda(unit)
             else:
                 component._handle_cuda(unit)
-                self.assertEqual(unit['description']['environment']['CUDA_VISIBLE_DEVICES'],
-                                 result)
+                unit_env = unit['description']['environment']
+                if result == 'KeyError':
+                    with pytest.raises(KeyError):
+                        self.assertIsNone(unit_env['CUDA_VISIBLE_DEVICES'])
```
Comment (iparask) [inline tests/test_scheduler/test_unit/test_base.py:114]: What is this line supposed to test? Can you provide some more information?

File: tests/test_scheduler/test_unit/test_base.py [hunk 2]
```diff
@@ -108,8 +108,12 @@ def test_handle_cuda(self,mocked_init):
                     component._handle_cuda(unit)
             else:
                 component._handle_cuda(unit)
-                self.assertEqual(unit['description']['environment']['CUDA_VISIBLE_DEVICES'],
-                                 result)
+                unit_env = unit['description']['environment']
+                if result == 'KeyError':
+                    with pytest.raises(KeyError):
```
Comment (iparask) [inline tests/test_scheduler/test_unit/test_base.py:113]:
```suggestion
                    with self.assertRaises(KeyError):
```

**Review by iparask — COMMENTED** [review]
Everything else looks good to me.

File: tests/test_scheduler/test_unit/test_base.py (hunk 1 above)
Comment (mtitov) [reply tests/test_scheduler/test_unit/test_base.py:115]: accessing the key `'CUDA_VISIBLE_DEVICES'` will raise an exception. Initially I did `_ = unit_env['CUDA_VISIBLE_DEVICES']`, but flake gave a warning, that's why such construction was chosen

**Pull request merged**